File: 总题库/1014.BestSightseeingPair.cpp
// 枚举法
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& A) {
        int mx = A[0] + 0;
        int res = INT_MIN;
        for (int i = 1; i < A.size(); ++i) {
            res = max(res, mx + A[i] - i);
            mx = max(mx, A[i] + i);
        }
        return res;
    }
};

File: 总题库/1014.BestSightseeingPair.py
class Solution:
    def maxScoreSightseeingPair(self, A: List[int]) -> int:
        mx = A[0] + 0
        res = float('-inf')
        for i in range(1, len(A)):
            res = max(res, mx + A[i] - i)
            mx = max(mx, A[i] + i)
        
        return res